### PR TITLE
applications: nrf_desktop: dfu_mcumgr: use boot info hook with suit

### PR DESCRIPTION
**Pull request by kapi-no**
Use the bootloader information hook for the nRF Desktop targets that use the SUIT DFU. The hook is used to override the default MCUmgr handler intended for MCUboot and to indicate the SUIT-based DFU.

Ref: NCSDK-29287

Follow-up PR after the following change in SUITFU:

https://github.com/nrfconnect/sdk-nrf/pull/17301

### CODE REVIEW

**Review by tejlmand — APPROVED** [review]

File: applications/nrf_desktop/doc/dfu_mcumgr.rst [hunk 1]
```diff
@@ -107,6 +107,10 @@ The SUIT backend configuration selects the following options:
 
 * :kconfig:option:`CONFIG_MGMT_SUITFU` - This option enables the SUIT Firmware Update Management that implements MCUmgr image management handlers required for the DFU process.
 * :kconfig:option:`CONFIG_MGMT_SUITFU_GRP_SUIT` - This option is needed to upload information about all manifests present on the device.
+* :kconfig:option:`CONFIG_MGMT_SUITFU_GRP_OS_BOOTLOADER_INFO_HOOK` - This option is needed to override the default MCUmgr handler intended for MCUboot bootloader information and replace it with the SUIT bootloader information.
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/doc/dfu_mcumgr.rst:110]: The `MGMT_SUITFU_GRP_OS_BOOTLOADER_INFO_HOOK` Kconfig depends on the `MCUMGR_GRP_OS_BOOTLOADER_INFO_HOOK` Kconfig - the misconfiguration is not possible

File: applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf [hunk 2]
```diff
@@ -133,7 +133,6 @@ CONFIG_MCUMGR_TRANSPORT_WORKQUEUE_STACK_SIZE=4096
 
 # Allow a DFU host to retrieve information about the number and size of MCUmgr buffers.
 # The nRF Connect Device Manager uses this information to speed up the DFU.
-CONFIG_MCUMGR_GRP_OS=y
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/configuration/nrf54h20dk_nrf54h20_cpuapp/prj.conf:136]: Well, we can further ask a question why `CONFIG_MCUMGR` is not explicitly enabled here, but I can change it

File: applications/nrf_desktop/doc/dfu_mcumgr.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/doc/dfu_mcumgr.rst:110]: Still, you can disable the `MGMT_SUITFU_GRP_OS_BOOTLOADER_INFO_HOOK` keeping the `MCUMGR_GRP_OS_BOOTLOADER_INFO_HOOK` enabled, right?

[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/doc/dfu_mcumgr.rst:110]: Or you can have `MCUMGR_GRP_OS_BOOTLOADER_INFO` enabled with no hook at all

**Comment by kapi-no** [comment]
Rebase-only operation due to the CI failures

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -299,6 +299,8 @@ nRF Desktop
   * The :ref:`nrf_desktop_dfu_mcumgr` to recognize the MCUmgr custom group ID (:kconfig:option:`CONFIG_MGMT_GROUP_ID_SUIT`) from the SUITFU subsystem (:kconfig:option:`CONFIG_MGMT_SUITFU`) as a DFU-related command group.
   * All build configurations with the DFU over MCUmgr support to require encryption for operations on the Bluetooth GATT SMP service (see the :kconfig:option:`CONFIG_MCUMGR_TRANSPORT_BT_PERM_RW_ENCRYPT` Kconfig option).
     The Bluetooth pairing procedure of the unpaired Bluetooth peers must now be performed before the DFU operation.
+  * The :ref:`nrf_desktop_dfu_mcumgr` to enable the bootloader information hook that is intended for SUIT-based DFU.
```
Comment (MarekPieta) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:302]: Please also inform that MCUboot has bootloader information hook enabled too

**Review by MarekPieta — CHANGES_REQUESTED** [review]

**Comment by kapi-no** [comment]
Rebasing, no file diff has changed

**Comment by kapi-no** [comment]
My next PR update will integrate minor changes that were introduced by @MarekPieta in the following PR:

https://github.com/nrfconnect/sdk-nrf/pull/17916

File: applications/nrf_desktop/doc/dfu_mcumgr.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/doc/dfu_mcumgr.rst:112]: Fixed here: https://github.com/nrfconnect/sdk-nrf/pull/17916

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 3 above)
Comment (kapi-no) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:332]: added a dedicated changelog for this part

File: applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr [hunk 4]
```diff
@@ -58,6 +59,7 @@ config DESKTOP_DFU_BACKEND_SUIT
 	select MCUMGR_SMP_LEGACY_RC_BEHAVIOUR
 	select MGMT_SUITFU
 	select MGMT_SUITFU_GRP_SUIT
+	select MGMT_SUITFU_GRP_OS_BOOTLOADER_INFO_HOOK
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.dfu_mcumgr:62]: The select is no longer needed (the Kconfig is enabled by default). Whole point of my change added to SUITFU was to make it work out-of-the-box with no extra configuration on application side needed.

**Review by MarekPieta — CHANGES_REQUESTED** [review]

**Review by MarekPieta — APPROVED** [review]

File: applications/nrf_desktop/doc/dfu_mcumgr.rst [hunk 5]
```diff
@@ -36,6 +36,13 @@ The module selects the following configurations:
 * :kconfig:option:`CONFIG_MCUMGR` - This option enables the MCUmgr support, which is required for the DFU process.
   For details, see :ref:`zephyr:mcu_mgr` in the Zephyr documentation.
 
+* :kconfig:option:`CONFIG_MCUMGR_GRP_OS` - This option enables the MCUmgr OS management handlers that are required for the DFU process.
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply applications/nrf_desktop/doc/dfu_mcumgr.rst:39]: done

**Comment by kapi-no** [comment]
Update the commit description without any file changes.

**Review by mkapala-nordic — APPROVED** [review]

**Pull request merged**